plugins/chanbackup.c

```diff
@@ -0,0 +1,329 @@
+#include "config.h"
+#include <ccan/array_size/array_size.h>
+#include <ccan/read_write_all/read_write_all.h>
+#include <ccan/noerr/noerr.h>
+#include <ccan/err/err.h>
+#include <ccan/fdpass/fdpass.h>
+#include <ccan/tal/str/str.h>
+#include <ccan/json_out/json_out.h>
+#include <ccan/cast/cast.h>
+#include <ccan/mem/mem.h>
+#include <ccan/str/hex/hex.h>
+#include <common/json_tok.h>
+#include <common/json.h>
+#include <common/json_stream.h>
+#include <common/hsm_encryption.h>
+#include <common/type_to_string.h>
+#include <plugins/libplugin.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sodium.h>
+
+/* Global secret object to keep the derived encryption key for the SCB */
+struct secret secret;
+
+/* Helper to fetch out SCB from the RPC call */
+static bool json_to_str (const char *buffer, const jsmntok_t *tok,
```

Comment:
by not just return the string?

Reply:
Tried to follow the convention from `json_helpers.c`. It's done in this way there...